src/mono/mono/component/debugger-agent.c

```diff
@@ -6433,7 +6429,7 @@ get_types_for_source_file (gpointer key, gpointer value, gpointer user_data)
 			files = get_source_files_for_type (klass);
 			g_hash_table_insert (info->source_files, klass, files);
 
-			for (int i = 0; i < files->len; ++i) {
+			for (guint i = 0; i < files->len; ++i) {
```

Comment:
Yes we probably could, did a quick test and it still builds without additional warnings. Any particular reason to do that, except that we might have usage in many places that do int i when iterating it? It will also affect the signature of g_ptr_array_remove_* that currently accept indexes to go from guint to int.

Reply:
Changing it to int would reduce the amount of changes required and make it easier to write code to iterate on it.

Reply:
So for the warnings, this PR makes sure all builds as expected, but enabling more warnings might trigger more warnings related to doing unsinged->signed conversion. This PR is already very big, and I mainly did small number of changes like the one above, just to align with original types in files already touched. Could an alternative be to look into that when we enable some additional warnings that will trigger errors due to singed/unsigned mismatch? An alternative would be to revert all of the int i -> source type in this PR and then take care of all in separate PR when enabling more warnings where we also could discuss if we should change type when iterating or at the source. So mainly three options for this PR:

* Leave as is, changes done so far will simplify future fixes of warnings, but might need to be change back in a couple of places if we change any of the source types like len -> int.
* Revert the int i -> source type done in this PR (unless it breaks some other warnings enabled by this PR).
* Change len to int and fix all usage of it as part of this PR, includes changing some of the array API's as well, but as said, this PR is already big enough.

Thoughts?